app.py

```diff
@@ -12,17 +12,17 @@ def validate(keyboard, pin):
     :return: nothing or raises ValueError if validation fails
     """
     if not all([isinstance(keyboard, list), isinstance(pin, str)]):
-        raise ValueError(
+        raise TypeError(
             f"Input is not valid. Keyboard must be a list. Pin must be a string."
         )
-    if not pin.isdigit():
-        raise ValueError(f"Input is not valid. Pin must be a string of digits.")
-    if not all([isinstance(el, list) for el in keyboard]) or not all(
-        [isinstance(el[i], str) for el in keyboard for i in range(len(el))]
-    ):
-        raise ValueError(
-            f"Input is not valid. Keyboard must be a list of lists of strings."
-        )
+    if keyboard == [] or pin == "":
```

Comment:
checking for emptiness is widely accepted as checking for *falsy* values, e.g. `if not keyboard:` or `if not pin`

Reply:
I usually do it like that `if not x: ...` but I wanted to bring up the problem for these types of cases at our next talk:
For this case if the keyboard input would be some other falsy value( 0, {}, "", None, etc) not an empty list. I wanted it to fall under the TypeError, and to only have ValueError for empty lists or strings.

Reply:
Fair point. What if you check the type before the value ?

Reply:
Yes, I will follow that order! :)

Reply:
Pun intended, hope it's not unprofessional for PRs